sklearn/ensemble/gradient_boosting_cv.py

```diff
@@ -0,0 +1,804 @@
+        fitting. Auto mode by default will use presorting on dense data and
+        default to normal sorting on sparse data. Setting presort to true on
+        sparse data will raise an error. If a list is given, each item in the
+        list will be evaluated.
+
+    Attributes
+    ----------
+    grid_scores_ : list of named tuples
+        Contains scores for all parameter combinations in param_grid.
+        Each entry corresponds to one parameter setting.
+        Each named tuple has the attributes:
+
+            * ``parameters``, a dict of parameter settings
+            * ``mean_validation_score``, the mean score over the
+              cross-validation folds
+            * ``cv_validation_scores``, the list of scores for each fold
+
+    best_score_tuple_ : named tuple
+        The attributes of the model which gave the best score.
+
+        * ``parameters``, a dict of parameter settings, which includes the
+          `n_estimators` parameter
+        * ``mean_validation_score``, the mean score over the
+          cross-validation folds
+        * ``cv_validation_scores``, the list of scores for each fold
+
+    best_params_ : dict
+        The parameters which gave the best score, along with the `n_estimators`
+        parameter.
+
+    best_estimator_ : GradientBoostingClassifier
+        If `refit` was `True`, the model with the best score fit to the entire
+        dataset.
+
+    """
+
+    _estimator_class = GradientBoostingClassifier
+
+    def __init__(self, n_stop_rounds=10, score_precision=2,
+                 max_iterations=10000, cv=3, n_iter_combo=None, scoring=None,
+                 refit=True, n_jobs=1, pre_dispatch='2*n_jobs', verbose=0,
+                 loss='deviance', learning_rate=0.1, subsample=1.0,
+                 min_samples_split=2, min_samples_leaf=1,
+                 min_weight_fraction_leaf=0., max_depth=3, init=None,
+                 random_state=None, max_features=None, max_leaf_nodes=None,
+                 presort='auto'):
+
+        self.n_stop_rounds = n_stop_rounds
+        self.score_precision = score_precision
+        self.max_iterations = max_iterations
+        self.cv = cv
+        self.n_iter_combo = n_iter_combo
+        self.scoring = scoring
+        self.refit = refit
+        self.n_jobs = n_jobs
+        self.pre_dispatch = pre_dispatch
+        self.verbose = verbose
+
+        self.loss = loss
+        self.learning_rate = learning_rate
+        self.subsample = subsample
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.min_weight_fraction_leaf = min_weight_fraction_leaf
+        self.max_depth = max_depth
+        self.init = init
+        self.random_state = random_state
+        self.max_features = max_features
+        self.max_leaf_nodes = max_leaf_nodes
+        self.presort = presort
+
+    def fit(self, X, y):
+        """Run fit with all sets of parameters till convergence.
```

Comment:
Check the docstring of the other CV estimators. Shouldn't this mention cross-validation?